### PR TITLE
libcephfs: fixup possible overflow

### PR DESCRIPTION
**Pull request by renhwztetecs**
including：
fixup get object_size way
fixup get pool_id way
fixup get stripe_count&& stripe_unit way

fix the lowest cost, in front of the not fully use these function interfaces

Signed-off-by: huanwen ren <ren.huanwen@zte.com.cn>

### CODE REVIEW

File: src/test/libcephfs/test.cc [hunk 1]
```diff
@@ -991,10 +991,17 @@ TEST(LibCephFS, BadFileDesc) {
 
   struct sockaddr_storage addr;
   ASSERT_EQ(ceph_get_file_stripe_address(cmount, -1, 0, &addr, 1), -EBADF);
+<<<<<<< HEAD
```
Comment (xiexingguo) [inline src/test/libcephfs/test.cc]: Don't know how  does it even get compiled...

Comment (renhwztetecs) [inline src/test/libcephfs/test.cc]: thanks

**Comment by renhwztetecs** [comment]
retest this please

**Comment by jcsp** [comment]
As @xiexingguo pointed out, there is broken code in here that won't compile.  Note that the automated PR check does not compile everything.

**Comment by renhwztetecs** [comment]
it's done
Have modified the test does not check?

**Comment by jcsp** [comment]
This still doesn't build: please fix and check that you can compile ceph_test_libcephfs

**Comment by jcsp** [comment]
```
cd "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu" && /usr/bin/cmake -E cmake_depends "Unix Makefiles" "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539" "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/test/libcephfs" "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu" "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/test/libcephfs" "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/DependInfo.cmake" --color=
Scanning dependencies of target ceph_test_libcephfs
make[4]: Leaving directory '/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu'
make -f src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/build.make src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/build
make[4]: Entering directory '/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu'
[ 86%] Building CXX object src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/test.cc.o
cd "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/test/libcephfs" && /usr/bin/aarch64-linux-gnu-g++ -DCEPH_LIBDIR=\"/usr/lib\" -DCEPH_PKGLIBDIR=\"/usr/lib/ceph\" -D__linux__ -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/include" -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src" -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/include" -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/xxHash" -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/googletest/googlemock/include" -I"/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/googletest/googletest/include" -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -rdynamic -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -ftemplate-depth-1024 -Wno-invalid-offsetof -Wnon-virtual-dtor -Wstrict-null-sentinel -Woverloaded-virtual -fdiagnostics-color=auto -I/usr/include/nss -I/usr/include/nspr -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fPIE -DHAVE_CONFIG_H -D__CEPH__ -D_REENTRANT -D_THREAD_SAFE -D__STDC_FORMAT_MACROS -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -I/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/googletest/googlemock/include -I/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/googletest/googlemock/include -I/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/googletest/googletest/include -I/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu/src/googletest/googletest/include -fno-strict-aliasing -std=c++11 -o CMakeFiles/ceph_test_libcephfs.dir/test.cc.o -c "/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/test/libcephfs/test.cc"
/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/test/libcephfs/test.cc: In member function ‘virtual void LibCephFS_UseUnmounted_Test::TestBody()’:
error: /srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/test/libcephfs/test.cc:1194:11: redeclaration of ‘int64_t pool_id’
int64_t pool_id;
^
/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/src/test/libcephfs/test.cc:1185:11: note: ‘int64_t pool_id’ previously declared here
int64_t pool_id;
^
src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/build.make:65: recipe for target 'src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/test.cc.o' failed
make[4]: *** [src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/test.cc.o] Error 1
make[4]: Leaving directory '/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.0-2234-g18cc539/obj-aarch64-linux-gnu'
CMakeFiles/Makefile2:13230: recipe for target 'src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/all' failed
make[3]: *** [src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
```

**Comment by renhwztetecs** [comment]
@jcsp 
compile pass in my local environment
...
Scanning dependencies of target ceph_test_libcephfs
[ 77%] Building CXX object src/test/libcephfs/CMakeFiles/ceph_test_libcephfs.dir/test.cc.o
[ 77%] Linking CXX executable ../../../bin/ceph_test_libcephfs
[ 78%] Built target ceph_test_libcephfs
[ 79%] Linking CXX executable ../../../bin/ceph_test_rados_api_stat
....
[ 99%] Built target cls_kvs
[ 99%] Linking CXX executable ../../bin/ceph_rgw_jsonparser
[ 99%] Built target ceph_rgw_jsonparser
[ 99%] Linking CXX executable ../../bin/ceph_rgw_multiparser
[ 99%] Built target ceph_rgw_multiparser
[ 99%] Linking CXX executable ../../bin/radosgw
[100%] Built target radosgw
[100%] Linking CXX executable ../../bin/radosgw-admin
[100%] Built target radosgw-admin
[100%] Linking CXX executable ../../bin/radosgw-token
[100%] Built target radosgw-token
[100%] Linking CXX executable ../../bin/radosgw-object-expirer
[100%] Built target radosgw-object-expirer
[100%] manpages building
[100%] Built target manpages

**Comment by renhwztetecs** [comment]
@jcsp ping

**Comment by jcsp** [comment]
So I hadn't really looked at the code so far (because of the build issues), but now that I do I'm not sure that changing these interfaces is wise.  Things like Samba and NFS Ganesha depend on them.

The cases where you're worried about overflow seem a bit unlikely to occur with sensible values -- I think I would be okay with just asserting out if we see e.g. a pool ID greater than 2 billion.

**Comment by renhwztetecs** [comment]
@jcsp 
thanks for your reply
I have see use assert way to deal with pool_id greater than 2^31, and others.
If the NFS and samba in used and modify the interface is not very wise.
assert is good idea.  I will close this PR  and  pull new PR

**Pull request closed without merging**